toolkit/smart-contracts/offchain/src/governed_map/mod.rs

```diff
@@ -316,3 +316,148 @@ pub async fn run_insert_with_force<
 	}
 	Ok(tx_hash_opt)
 }
+
+pub async fn run_update<
+	C: QueryLedgerState + QueryNetwork + Transactions + QueryUtxoByUtxoId,
+	A: AwaitTx,
+>(
+	genesis_utxo: UtxoId,
+	key: String,
+	value: ByteString,
+	expected_current_value: Option<ByteString>,
+	payment_signing_key: &CardanoPaymentSigningKey,
+	ogmios_client: &C,
+	await_tx: &A,
+) -> anyhow::Result<Option<MultiSigSmartContractResult>> {
+	let ctx = TransactionContext::for_payment_key(payment_signing_key, ogmios_client).await?;
+	let (validator, policy) = crate::scripts_data::governed_map_scripts(genesis_utxo, ctx.network)?;
+	let validator_address = validator.address_bech32(ctx.network)?;
+	let validator_utxos = ogmios_client.query_utxos(&[validator_address]).await?;
+
+	let Some(actual_current_value) =
+		get_current_value(validator_utxos.clone(), key.clone(), policy.policy_id())
+	else {
+		return Err(anyhow!("Cannot update nonexistent key :'{key}'."));
+	};
+
+	if matches!(expected_current_value,
+		Some(ref expected_current_value) if *expected_current_value != actual_current_value)
+	{
+		return Err(anyhow!(
+				"Value for key '{key}' is set to a different value than expected. Actual: {actual_current_value:?} Expected: {expected_current_value:?}"
+			));
+	}
+
+	let tx_hash_opt = {
+		if actual_current_value != value {
+			Some(
+				update(
+					&validator,
+					&policy,
+					key,
+					value,
+					&validator_utxos,
+					ctx,
+					genesis_utxo,
+					ogmios_client,
+				)
+				.await?,
+			)
+		} else {
+			log::info!("Value for key '{key}' is already set to {actual_current_value:?}. Skipping update.");
```

Comment:
Cosmetic: it will display: "... is set to ByteString(<some hex>). Skipping ...". I mean, we throw Debug format at users.